### ISSUE TITLE
Training fails with zero Gaussian instances on A100

### ISSUE THREAD

**Issue by WanqiYuan** [yimgshao/refracgs#1]
Hi!

Thank you for sharing this great work!

We encountered an issue when training RefracGS on an NVIDIA A100 using the default configuration. We tested both nerfrac_real.yaml and
  refracgs_synthetic_30view.yaml. Training starts normally, but around step 600 all 20,000 Gaussians are pruned, the loss becomes NaN, and OptiX repeatedly reports:

  "buildInputs[0].instanceArray", "numInstances" is zero
  "tempBufferSizeInBytes" is non-zero but "tempBuffer" is null
  outputBuffer is 0

Before pruning, the reconstruction appears nearly black and the opacity continuously decreases, suggesting that the default primitive_type: instances path may not be producing valid ray–Gaussian intersections or gradients.

Our environment:
  - NVIDIA A100
  - CUDA 11.8
  - PyTorch 2.7.1+cu118

Do you know what might cause this behavior? Are there specific OptiX/CUDA/PyTorch versions required, or a recommended fix for the default instance-based renderer? Is there any parameter that needs to be changed?

Thanks for your help!

The following are error information:
Training ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╺━━━━━━  83% step: 834 loss: nan :: 🚗💨 Speed: 50.11it/s :: 🕒 Elapsed: 0:00:16[ 2][       ACCEL]: "tempBufferSizeInBytes" is non-zero but "tempBuffer" is null
[ 2][       ACCEL]: "buildInputs[0].instanceArray", "numInstances" is zero, but "instances" is non-null
[ 2][       ERROR]: outputBuffer is 0
[ 2][       ACCEL]: "tempBufferSizeInBytes" is non-zero but "tempBuffer" is null
[ 2][       ACCEL]: "buildInputs[0].instanceArray", "numInstances" is zero, but "instances" is non-null
[ 2][       ERROR]: outputBuffer is 0
[ 2][       ACCEL]: "tempBufferSizeInBytes" is non-zero but "tempBuffer" is null
[ 2][       ACCEL]: "buildInputs[0].instanceArray", "numInstances" is zero, but "instances" is non-null
[ 2][       ERROR]: outputBuffer is 0
[ 2][       ACCEL]: "tempBufferSizeInBytes" is non-zero but "tempBuffer" is null
[ 2][       ACCEL]: "buildInputs[0].instanceArray", "numInstances" is zero, but "instances" is non-null
[ 2][       ERROR]: outputBuffer is 0
[ 2][       ACCEL]: "tempBufferSizeInBytes" is non-zero but "tempBuffer" is null
[ 2][       ACCEL]: "buildInputs[0].instanceArray", "numInstances" is zero, but "instances" is non-null
[ 2][       ERROR]: outputBuffer is 0

**Comment by yimgshao** [comment]
Hi,

Thank you for your interest in our work and for reporting this issue!

We have only tested RefracGS on NVIDIA RTX 4090 GPUs, so unfortunately we cannot confirm the exact cause of this issue. However, we suspect that it may be related to the A100 not having dedicated RT Cores, which are used by the 3DGRT ray-tracing pipeline.

A similar issue with 3DGRUT on A100 GPUs has also been reported here:
https://github.com/nv-tlabs/3dgrut/issues/30

We would recommend using an NVIDIA GPU with RT Cores (e.g., RTX 4090) if possible.

Thanks again for bringing this to our attention!
